docs/proposals/forecasting/20230613-framework-of-prediction-and-detection.md

```diff
@@ -0,0 +1,289 @@
+time, so prediction of task metric is valuable when routing task from central.
+
+These topics are all related to data analysis, so we need a general framework for prediction and detection to support 
+different scenarios. 
+
+### Goals
+- Design a holistic framework with modules of profile control, forecasting and data scraping to support data prediction and detection.
+
+- Clarify the data type and transfer links, and how to save the storage and collection overhead.
+
+- Collaboration and division of builtin and self-extended metric exporters, and adaption of multiple using scenarios.
+
+### Non-Goals
+- Design of forecasting algorithm. The interference detection and resource prediction may share same prediction models,
+however, this is over-detailed now. 
+
+- Supports of MPA framework proposed in Kubernetes autoscaler. Although our work can run as a recommender for MPA after
+some protocol conversion works, we can discuss in the following proposals.
+
+- Long-term optimization, even though we have already mentioned some ideas about saving transfer cost and storage overhead.  
+
+## Proposal
+
+### User Stories
+
+#### Detect Interference for Noisy Neighbor Problem
+Numbers of applications running in K8s cluster, the cluster administrator needs to know whether some pods got
+interference by others through general methods, instead of application-specified metrics. Advanced metrics like
+CPI, PSI are provided for analysis and modeling, and interference will be detected once there is an outlier. 
+  
+
+#### Estimation of Resource Consumption
+Collecting history metrics to estimate resource usage of containers or tasks, which can help the right-sizing and scheduling
+optimization for load-balancing.
+
+#### Time-series Data Prediction
+HPA and VPA control the scaling actions of K8s workloads, which rely on the prediction of resource usage over time. We can
+also customize the recommender with different algorithms on the [autoscaler](https://cloud.google.com/kubernetes-engine/docs/concepts/horizontalpodautoscaler?hl=zh-cn) 
+as needed. 
+
+
+## Degign Details
+
+### Which metric API should we use
+Kubernetes has defined three types of metric api: `metrics.k8s.io`, `custom.metrics.k8s.io` and `external.metrics.k8s.io`,
+which can be accessed as Kubernetes resources.
+
+`Prometheus` is another widely used metric service, which also provides good extendability. User can collect metrics in their
+own components(e.g. daemon-set, operator), and expose as standard exporter format. Then these metrics can be accessed from
+`Prometheus` after the service monitor defined.
+
+Besides, Kubernetes also provides customized API as CRD and aggregate layer in `apiserver`, which can be used for metric
+report. Furthermore, we can define dedicated protocols as an internal metric service.
+
+`Prometheus` seems to be a good option before we take a deep look at the pros and cons.
+
+Pros:
```

Comment:
Promtheus Metrics can be adapted to Kubernetes Metric API by Prometheus Adapter. Thus we don't need a metric server.